### DISCUSSION TITLE
P2 UI and E4

### DISCUSSION THREAD

**Discussion by laeubi** [eclipse-equinox/p2#83] [General]
Currently the P2 UI is heavily bound to the E3 model, mostly because of excessive usage of `PlatformUI.getWorkbench()`, this currently makes it impossible to use the P2 UI in a 'pure' E4 RCP application. Because the P2 UI is non trivial, reimplementing it is most of the time not an option.

Even though it might be much work, I wonder if smaller steps towards this goal would be desired/accepted by P2.

**Comment by akurtakov** [comment]
On Sun, Jun 12, 2022, 9:18 PM Christoph Läubrich ***@***.***>
wrote:

> Currently the P2 UI is heavily bound to the E3 model, mostly because of
> excessive usage of PlatformUI.getWorkbench(), this currently makes it
> impossible to use the P2 UI in a 'pure' E4 RCP application. Because the P2
> UI is non trivial, reimplementing it is most of the time not an option.
>
> Even though it might be much work, I wonder if smaller steps towards this
> goal would be desired/accepted by P2
>

IMHO, migrating more and more parts to e4 is more than welcome and actually
we are quite late with this.
